## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-fs.sh | :x: [more...](#0b468161-460b-4253-8b79-c63c2919b483) |
<a name="0b468161-460b-4253-8b79-c63c2919b483"></a>
### 7374-31e4fd1/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:30443
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 806a493ba464:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 46 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 24 seconds
(8/14) Running minio-dotnet tests ... done in 24 seconds
(9/14) Running minio-go tests ... done in 34 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 32 seconds
(11/14) Running minio-js tests ... done in 36 seconds
(12/14) Running minio-py tests ... FAILED in 2 minutes and 21 seconds
{
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "duration": 41,
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "message": "AccessDenied: message: Access Denied",
  "args": {
    "object_name": "ed6fab73-f99d-4ca0-bc7e-2cffed6fa750",
    "bucket_name": "minio-py-test-69cc1a95-c8a9-4fe1-ac28-3244d9616cda"
  },
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```

--- a/cmd/globals.go
+++ b/cmd/globals.go
@@ -57,7 +57,6 @@ const (
 	globalMinioDefaultStorageClass = "STANDARD"
 	globalWindowsOSName            = "windows"
 	globalNetBSDOSName             = "netbsd"
-	globalSolarisOSName            = "solaris"
 	globalMinioModeFS              = "mode-server-fs"
 	globalMinioModeXL              = "mode-server-xl"
 	globalMinioModeDistXL          = "mode-server-distributed-xl"

--- a/cmd/posix.go
+++ b/cmd/posix.go
@@ -232,7 +232,6 @@ func getDiskInfo(diskPath string) (di disk.Info, err error) {
 var ignoreDiskFreeOS = []string{
 	globalWindowsOSName,
 	globalNetBSDOSName,
-	globalSolarisOSName,
 }
 
 // check if disk total has minimum required size.
@@ -943,20 +942,6 @@ func (s *posix) openFile(volume, path string, mode int) (f *os.File, err error) 
 	return w, nil
 }
 
-// Just like io.LimitedReader but supports Close() to be compatible with io.ReadCloser that is
-// returned by posix.ReadFileStream()
-type posixLimitedReader struct {
-	io.LimitedReader
-}
-
-func (l *posixLimitedReader) Close() error {
-	c, ok := l.R.(io.Closer)
-	if !ok {
-		return errUnexpected
-	}
-	return c.Close()
-}
-
 // ReadFileStream - Returns the read stream of the file.
 func (s *posix) ReadFileStream(volume, path string, offset, length int64) (io.ReadCloser, error) {
 	var err error
@@ -1030,7 +1015,10 @@ func (s *posix) ReadFileStream(volume, path string, offset, length int64) (io.Re
 	if _, err = file.Seek(offset, io.SeekStart); err != nil {
 		return nil, err
 	}
-	return &posixLimitedReader{io.LimitedReader{R: file, N: length}}, nil
+	return struct {
+		io.Reader
+		io.Closer
+	}{Reader: io.LimitReader(file, length), Closer: file}, nil
 }
 
 // CreateFile - creates the file.

--- a/cmd/storage-rest-server.go
+++ b/cmd/storage-rest-server.go
@@ -47,6 +47,7 @@ type storageRESTServer struct {
 func (s *storageRESTServer) writeErrorResponse(w http.ResponseWriter, err error) {
 	w.WriteHeader(http.StatusForbidden)
 	w.Write([]byte(err.Error()))
+	w.(http.Flusher).Flush()
 }
 
 // Authenticates storage client's requests and validates for skewed time.
@@ -93,6 +94,7 @@ func (s *storageRESTServer) GetInstanceID(w http.ResponseWriter, r *http.Request
 	}
 	w.Header().Set("Content-Length", strconv.Itoa(len(s.instanceID)))
 	w.Write([]byte(s.instanceID))
+	w.(http.Flusher).Flush()
 }
 
 // DiskInfoHandler - returns disk info.
@@ -268,6 +270,7 @@ func (s *storageRESTServer) ReadAllHandler(w http.ResponseWriter, r *http.Reques
 	}
 	w.Header().Set("Content-Length", strconv.Itoa(len(buf)))
 	w.Write(buf)
+	w.(http.Flusher).Flush()
 }
 
 // ReadFileHandler - read section of a file.
@@ -311,6 +314,7 @@ func (s *storageRESTServer) ReadFileHandler(w http.ResponseWriter, r *http.Reque
 	}
 	w.Header().Set("Content-Length", strconv.Itoa(len(buf)))
 	w.Write(buf)
+	w.(http.Flusher).Flush()
 }
 
 // ReadFileHandler - read section of a file.
@@ -331,6 +335,7 @@ func (s *storageRESTServer) ReadFileStreamHandler(w http.ResponseWriter, r *http
 		s.writeErrorResponse(w, err)
 		return
 	}
+
 	rc, err := s.storage.ReadFileStream(volume, filePath, int64(offset), int64(length))
 	if err != nil {
 		s.writeErrorResponse(w, err)
@@ -338,7 +343,9 @@ func (s *storageRESTServer) ReadFileStreamHandler(w http.ResponseWriter, r *http
 	}
 	defer rc.Close()
 	w.Header().Set("Content-Length", strconv.Itoa(length))
+
 	io.Copy(w, rc)
+	w.(http.Flusher).Flush()
 }
 
 // ListDirHandler - list a directory.

